### PR TITLE
LibJS+LibRegex: Some parser fixes

### PR DESCRIPTION
**Pull request by davidot**

### CODE REVIEW

File: Userland/Libraries/LibC/regex.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ enum __Regex_Error {
     __Regex_InvalidCaptureGroup,        // Content of capture group is invalid.
     __Regex_InvalidNameForCaptureGroup, // Name of capture group is invalid.
     __Regex_InvalidNameForProperty,     // Name of property is invalid.
+    __Regex_DuplicateNamedCapture,      // Duplicate named capture group
```
Comment (davidot) [inline Userland/Libraries/LibC/regex.h:41]: I only now realize that this is a LibC file, should all regex errors be added to this? All the other ones were so I just did it.

Comment (alimpfard) [reply Userland/Libraries/LibC/regex.h:41]: Yeah, that's fine.

**Comment by davidot** [comment]
The CI is failing because of it can't fetch stuff for the toolchain.
But the lagom NoFuzz tests do pass which includes the regex and libjs tests so it _should_ be safe.

File: Userland/Libraries/LibJS/Parser.cpp [hunk 2]
```diff
@@ -3905,17 +3912,70 @@ void Parser::check_identifier_name_for_assignment_validity(StringView name, bool
...
+    consume(TokenType::CurlyClose);
+}
+
 NonnullRefPtr<ImportStatement> Parser::parse_import_statement(Program& program)
 {
+    // We use the extended syntax which adds:
+    //  ImportDeclaration:
+    //      importImportClauseFromClause[no LineTerminator here]AssertClause;
+    //      importModuleSpecifier[no LineTerminator here]AssertClause;
+    // From:  https://tc39.es/proposal-import-assertions/#prod-ImportDeclaration
```
Comment (linusg) [inline Userland/Libraries/LibJS/Parser.cpp:3965]:
```suggestion
    //      import ImportClause FromClause [no LineTerminator here] AssertClause ;
    //      import ModuleSpecifier [no LineTerminator here] AssertClause ;
    // From: https://tc39.es/proposal-import-assertions/#prod-ImportDeclaration
```

File: Userland/Libraries/LibJS/Parser.cpp [hunk 3]
```diff
@@ -4028,11 +4091,16 @@ NonnullRefPtr<ImportStatement> Parser::parse_import_statement(Program& program)
...
-    return create_ast_node<ImportStatement>({ m_state.current_token.filename(), rule_start.position(), position() }, module_name, move(entries));
+    return create_ast_node<ImportStatement>({ m_state.current_token.filename(), rule_start.position(), position() }, move(module_request), move(entries));
 }
 
 NonnullRefPtr<ExportStatement> Parser::parse_export_statement(Program& program)
 {
+    // We use the extended syntax which adds:
+    //  ExportDeclaration:
+    //      exportExportFromClauseFromClause[no LineTerminator here] AssertClause;
+    // From:  https://tc39.es/proposal-import-assertions/#prod-ExportDeclaration
```
Comment (linusg) [inline Userland/Libraries/LibJS/Parser.cpp:4102]:
```suggestion
    // We use the extended syntax which adds:
    //  ExportDeclaration:
    //      export ExportFromClause FromClause [no LineTerminator here] AssertClause ;
    // From: https://tc39.es/proposal-import-assertions/#prod-ExportDeclaration
```

**Review by linusg — COMMENTED** [review]
Looks great!

File: Userland/Libraries/LibJS/Parser.cpp (hunk 2 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Parser.cpp]: Whoops yeah that's unreadable like that. Fixed!

File: Userland/Libraries/LibJS/Parser.cpp [hunk 4]
```diff
@@ -3912,17 +3912,70 @@ void Parser::check_identifier_name_for_assignment_validity(StringView name, bool
...
+    }
+
+    consume(TokenType::CurlyClose);
+}
+
 NonnullRefPtr<ImportStatement> Parser::parse_import_statement(Program& program)
 {
+    // We use the extended syntax which adds:
+    //  ImportDeclaration:
+    //      import ImportClauseFromClause [no LineTerminator here] AssertClause;
```
Comment (linusg) [inline Userland/Libraries/LibJS/Parser.cpp:3963]: needs one more space :^)

```suggestion
    //      import ImportClause FromClause [no LineTerminator here] AssertClause;
```

**Review by linusg — APPROVED** [review]

**Pull request merged**